### PR TITLE
Feature/csd 2

### PR DESCRIPTION
**Pull request by karensamara**
New feature filter by area

### CODE REVIEW

File: src/custom.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+declare module '*.svg' {
```
Comment (dcruzb) [inline src/custom.d.ts:1]: I believe this is fixed in another branch, this module should not be created.

File: src/pages/Home/index.tsx [hunk 2]
```diff
@@ -2,7 +2,21 @@ import { DeadlineProps } from '../../components/Conference';
...
+import { CheckedContextProvider } from '../../contexts/CheckedContext';
+
+const FilterContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  margin-top: 20px;
+`;
+
+const FilterByAreaContainer = styled.div`
+  width: 20%;
```
Comment (dcruzb) [inline src/pages/Home/index.tsx:17]: Please avoid using styles in the same file as the code

**Review by dcruzb — CHANGES_REQUESTED** [review]

File: src/contexts/CheckedContext.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// CheckedContext.tsx
+import React, { createContext, useState } from 'react';
+
+export interface CheckedContextType {
+  checked: string[];
```
[the remark this answers is not in the document]
Comment (karensamara) [reply src/contexts/CheckedContext.tsx:5]: The 'checked' state is being used to store the values of the filter by area checkboxes, not as a boolean. To represent the state more accurately, I have renamed it to 'checkedValues'.

File: src/pages/Home/styles.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components';
```
Comment (dcruzb) [inline src/pages/Home/styles.tsx:1]: There is no need to styled-components here

File: package.json [hunk 5]
```diff
@@ -42,6 +45,7 @@
     "@types/papaparse": "^5.3.7",
     "@types/react": "^18.1.0",
     "@types/react-dom": "^18.2.6",
+    "@types/styled-components": "^5.1.26",
```
Comment (dcruzb) [inline package.json:48]: There is no need to use styled-components, could use import { styled } from '@mui/material';

File: src/pages/Home/index.tsx [hunk 6]
```diff
@@ -52,21 +70,29 @@ function Home() {
...
         {'The top CS conferences are listed in '}
         <Link target='_blank' href='https://CSRankings.org'>
           CSRankings.org
         </Link>
       </Typography>
-      {loading && (
-        <Box display='flex' justifyContent='center' alignItems='center' flexGrow='1' width='100%'>
-          <CircularProgress />
-        </Box>
+      {loading && <CircularProgress />}
```
Comment (dcruzb) [inline src/pages/Home/index.tsx:85]: The circular progress need to be inside a div to be centralized

File: src/pages/Home/index.tsx [hunk 7]
```diff
@@ -52,21 +70,29 @@ function Home() {
...
+      {!loading && (
+        <FilterContainer>
```
Comment (dcruzb) [inline src/pages/Home/index.tsx:87]: The style need a revision since this messed with the layout

**Review by dcruzb — CHANGES_REQUESTED** [review]

File: src/pages/Home/index.tsx [hunk 8]
```diff
@@ -52,21 +70,29 @@ function Home() {
     });
   }, []);
 
+  const handleCheckedChange = (checkedChangedValues: string[]) => {
+    setCheckedValues(checkedChangedValues);
+  };
+
   return (
-    <>
+    <main>
```
Comment (dcruzb) [inline src/pages/Home/index.tsx:78]: This defaults to display: block, breaking the flex display from cinnamon

**Review by dcruzb — CHANGES_REQUESTED** [review]

**Review by dcruzb — APPROVED** [review]

**Pull request merged**